pkg/pillar/worker/workerpool.go

```diff
@@ -125,8 +125,30 @@ func (wp *Pool) NumWorkers() int {
 	return len(wp.workers)
 }
 
-// TrySubmit returns false if the number of workers is already at the max
-// returns JobInProgressError if a job with that key already in progress
+// Submit submits jobs to the WorkerPool. If it cannot find a worker in the pool
+// that can service it - i.e. both the number of workers is at the maximum and the
+// queues of all workers are full - then it waits one second and tries all available
+// workers again, in an infinite loop until it finds an available worker.
```

Comment:
This forever wait behavior isn't usable in EVE since it risk getting a watchdog due to a touch file not being touched.

Reply:
Is it any different than the `Worker.Submit()` we used before, where it waited?

Reply:
No, but I replaced that with using Worker.TrySubmit before adding the pool stuff.

Having an option to specify a max time to wait would make it more interesting to consider. Can set that to infinite or zero to mean wait forever.

Reply:
Or use context.Context so the caller can set a deadline, timeout and/or cancel?

Reply:
Does that fit within the scope of this PR, where we just are aligning them? The risk exists with `Submit()` on a single worker as well. Maybe a separate PR to add a context afterwards?

I don't disagree, just think it is a separate small change.

Reply:
Yeah we can do the context separately. 
Can the Submit function be common for Single and Pool? It shouldn't care which TrySubmit it calls.

Reply:
> Can the Submit function be common for Single and Pool? It shouldn't care which TrySubmit it calls.

I don't understand. Isn't it the same signature now?

Reply:
I think this is the only missing piece.

Reply:
I was thinking ahead of a future Submit which uses context. For that it would be good if we can have a single implementation of Submit which uses TrySubmit, then waits while using the context. But that isn't what we have today. So all good for now.

Reply:
Got it.